### ISSUE TITLE
Notifications on smartwatch don't contain message content

### ISSUE THREAD

**Issue by 1nikolas** [signalapp/Signal-Android#12961]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When syncing Signal's notifications to my smartwatch, I just get "Most recent from: \<contact name\>" instead of the message content. I have tried 2 different brand smartwatches and have the same problem so I figured this should be a problem in your end. Notifications on the phone appear fine, it's just what the smartwatch app reads from the system. I found a closed issue about this but it's too old (from the TextSecure days of the app #4038). Maybe it's time to fix this?

### Steps to reproduce
- Have Signal
- Have a smartwatch with Signal notifications synced
- Wait for a notification

**Actual result:** Notification content on the smartwatch is "Most recent from: \<contact name\>"
**Expected result:** Notification content should be the message itself

### Device info

**Device:** OnePlus 8T
**Android version:** 13
**Signal version:** 6.21.2

**Comment by nicholas-signal** [comment]
![signal-2023-05-22-163312](https://github.com/signalapp/Signal-Android/assets/112583042/4c75f48a-05b1-4c27-91a2-b9923c83a7e0)

I am unable to reproduce this issue on my Galaxy Watch5. Would you mind sharing what smartwatch you're using, and what operating system it is running?

**Comment by 1nikolas** [comment]
It's an OnePlus Watch. The other one is a generic Chinese one. I think this is a problem with non-WearOS devices

**Comment by nicholas-signal** [comment]
I see. Unfortunately, we only have Wear OS and Tizen devices to test with, but Signal uses standard Notification APIs across the board. Do you experience this behavior with other apps like WhatsApp and Telegram?

**Comment by 1nikolas** [comment]
No. To me it looks like Signal is sending that extra notification (or maybe android does?) and appears on top of the normal message notification. If that notification is not even displayed, what's its purpose?

**Comment by 1nikolas** [comment]
The notification looks like this (I used a notification log app to see it)
![Screenshot_2023-05-22-23-42-23-22_73e67ff144ea0c27af74c0b29b3d293a.jpg](https://github.com/signalapp/Signal-Android/assets/30593419/fc0faa9b-44dd-4f7f-adae-e87fdbeb7595)

**Comment by nicholas-signal** [comment]
Signal notifications work using a multi-step process:
1. The Signal service sends an encrypted push message over Google's push notification service. This message contains the bare minimum amount of data to make it to the phone.
2. Receiving this notification causes the Signal app to "wake up" on the phone, at which point it establishes a direct connection with the Signal service.
3. Once connected to the Signal service, it retrieves the any encrypted data that the service has ready for it.
4. Locally on the device, the Signal client decrypts the data. Some of it is notification-worthy (a new message) some of it is not (read receipts). For the notification worthy stuff, the app generates a local system notification and hands it off to the operating system.

**Comment by 1nikolas** [comment]
So this "most recent from" thing has a purpose? And what would be the solution to this? Tell smartwatch manufactures to manually block this notification on their app?

**Comment by nicholas-signal** [comment]
Can we just check one thing? What do you have this setting set to in your Signal app settings?
![Screenshot_20230522_171158](https://github.com/signalapp/Signal-Android/assets/112583042/f9969af0-e7f3-4c8a-85e5-da609d9de741)

**Comment by 1nikolas** [comment]
![Screenshot_2023-05-23-00-13-16-94_4b14b257287ac0a6d1fd5fbb2487716f.jpg](https://github.com/signalapp/Signal-Android/assets/30593419/2a718479-dad2-4d80-9fbc-25afce70ab02)

**Comment by nicholas-signal** [comment]
Would you mind submitting a debug log?

**Comment by 1nikolas** [comment]
Don't think there is something with my device specifically but sure
[signal-log-1684873784417.zip](https://github.com/signalapp/Signal-Android/files/11548254/signal-log-1684873784417.zip)

**Comment by nicholas-signal** [comment]
So upon digging into how this all works, I've determined that the Signal app shows a "summary notification" if there's more than one unread conversation. It then replaces this with individual conversation notifications. I'll confer with the team to better understand how this came to be, and how difficult it would be to rearchitect.

**Comment by 1nikolas** [comment]
Well that's unfortunate.. This is still a problem

**Comment by acolombier** [comment]
Hi, I am also a OnePlus Watch owner and been experiencing this too. As per asked above, other chatting apps like WhatsApp and Messenger displays just fine. I am however unsure if the non-Wear OS would be having some custom logic on it, e.g filtering out the summary notification. I have no experience in Kotlin but I'd be happy to try and help fixing this issue if you would have some suggestion on how to approach it. Do you reckon making the summary notification would bu sufficient or is there a deeper work required for this?
